controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -510,6 +510,26 @@ func (r *BareMetalHostReconciler) actionUnmanaged(prov provisioner.Provisioner,
 	return actionContinue{unmanagedRetryDelay}
 }
 
+func (r *BareMetalHostReconciler) actionDetached(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+	if info.host.OperationalStatus() != metal3v1alpha1.OperationalStatusDetached {
+		provResult, err := prov.Detach()
+		if err != nil {
+			return actionError{errors.Wrap(err, "failed to detach")}
+		}
+		if provResult.ErrorMessage != "" {
+			return recordActionFailure(info, metal3v1alpha1.DetachError, provResult.ErrorMessage)
+		}
+		if provResult.Dirty {
+			return actionContinue{provResult.RequeueAfter}
+		}
+		info.log.Info("host is detached, removed from provisioner")
+	}
+	if info.host.SetOperationalStatus(metal3v1alpha1.OperationalStatusDetached) {
+		return actionUpdate{}
```

Comment:
This will requeue immediately, which is fine but we don't actually need to do that. We could return `actionUpdate{actionContinue{unmanagedRetryDelay}}`.

Reply:
Done!